src/components/mols/UserBar/index.tsx

```diff
@@ -18,22 +37,121 @@ const BarBadge = ({ value, icon }: BadgeProps) => {
     </div>
   );
 };
-
-const UserBar = () => {
+const MenuToggle = (component: any, setcomponent: any) => {
+  if (component === false) {
+    setcomponent(true);
+  } else {
+    setcomponent(false);
+  }
```

Comment:
```suggestion
  setcomponent(prevValue => !prevValue);
```

Reply:
Not sure why this function is outside the component. Why are we sending state (component) and setState (setComponent) instead of having this function INSIDE the component so it will be able to use the `const [menuOpen, setMenuOpen] = useState(false);`